packages/mui-base/src/useList/useList.test.tsx

```diff
@@ -85,4 +85,33 @@ describe('useList', () => {
       }),
     );
   });
+
+  describe('external props', () => {
+    it('should pass arbitrary props to the root slot', () => {
+      const handleClick = spy();
+
+      function Listbox() {
+        const { getRootProps } = useList({
+          items: [],
+          getItemId: () => undefined,
+        });
+        return (
+          <div
+            role="listbox"
+            {...getRootProps({ 'data-testid': 'test-listbox', onClick: handleClick })}
+          />
+        );
+      }
+
+      const { getByRole } = render(<Listbox />);
+
+      const listbox = getByRole('listbox');
+
+      expect(listbox).to.have.attribute('data-testid', 'test-listbox');
+
+      fireEvent.click(listbox);
+
+      expect(handleClick.callCount).to.equal(1);
```

Comment:
Nit: could we condense it a bit?

```suggestion
      const { getByRole } = render(<Listbox />);

      const listbox = getByRole('listbox');
      expect(listbox).to.have.attribute('data-testid', 'test-listbox');

      fireEvent.click(listbox);
      expect(handleClick.callCount).to.equal(1);
```